src/frontend/js/components/AshleyEditor/index.spec.tsx

```diff
@@ -31,6 +38,105 @@ describe('AshleyEditor', () => {
     const linkButton = container.querySelector(
       'path[d="M3.9 12c0-1.71 1.39-3.1 3.1-3.1h4V7H7c-2.76 0-5 2.24-5 5s2.24 5 5 5h4v-1.9H7c-1.71 0-3.1-1.39-3.1-3.1zM8 13h8v-2H8v2zm9-6h-4v1.9h4c1.71 0 3.1 1.39 3.1 3.1s-1.39 3.1-3.1 3.1h-4V17h4c2.76 0 5-2.24 5-5s-2.24-5-5-5z"]',
     );
+
     expect(linkButton).toBeInTheDocument();
+
+    // check if code button has been added to the toolbar plugin editor
+    const codeButton = container.querySelector(
+      'path[d="M9.4 16.6L4.8 12l4.6-4.6L8 6l-6 6 6 6 1.4-1.4zm5.2 0l4.6-4.6-4.6-4.6L16 6l6 6-6 6-1.4-1.4z"]',
+    );
+    expect(codeButton).toBeInTheDocument();
+  });
+
+  it('loads AshleyEditor component with existing json content containing a code-block and an unstyled block', () => {
+    const target = document.createElement('input');
+    target.value = BlockMapFactory([
+      {
+        key: 'brt6g',
+        text: 'ce texte est ajouté dans code-block.\n',
+        type: 'code-block',
+      },
+      {
+        type: 'unstyled',
+        text: 'ce texte est en gras.',
+        inlineStyleRanges: [
```

Comment:
can you order the key alphabetically ?

Reply:
I've managed to order these ones but couldn't order the ones in the factory as I'm comparing it after changing the editor and draft-js orders it in another way.